examples/basic_usage/main.tf

```diff
@@ -0,0 +1,6 @@
+module "observe_collection_example" {
+  source = "../../"
```

Comment:
Repeating for emphasis, this means the examples are broken on the Terraform Registry and cannot be copied. 

https://developer.hashicorp.com/terraform/language/modules/develop/structure

> Because examples will often be copied into other repositories for customization, any module blocks should have their source set to the address an external caller would use, not to a relative path.

To use examples as tests against HEAD, rewrite that external caller address to this relative path.